src/GraphQL/Types/ArgumentInformation.cs

```diff
@@ -0,0 +1,179 @@
+            : this(parameterInfo, sourceType, fieldType, typeInformation, null)
+        {
+            if (parameterInfo.ParameterType == typeof(IResolveFieldContext))
+            {
+                Expression<Func<IResolveFieldContext, IResolveFieldContext>> expr = x => x;
+                Expression = expr;
+            }
+            else if (parameterInfo.ParameterType == typeof(CancellationToken))
+            {
+                Expression<Func<IResolveFieldContext, CancellationToken>> expr = x => x.CancellationToken;
+                Expression = expr;
+            }
+        }
+
+        /// <summary>
+        /// The method parameter.
+        /// </summary>
+        public ParameterInfo ParameterInfo { get; }
+
+        /// <summary>
+        /// The expected type of <see cref="IResolveFieldContext.Source"/>.
+        /// Should equal <c>TSourceType</c> within <see cref="AutoRegisteringObjectGraphType{TSourceType}"/>.
+        /// </summary>
+        public Type SourceType { get; }
+
+        /// <summary>
+        /// The <see cref="Types.FieldType"/> that the query argument will be added to.
+        /// </summary>
+        public FieldType FieldType { get; }
+
+        /// <summary>
+        /// The parsed type information of the method parameter.
+        /// </summary>
+        public TypeInformation TypeInformation { get; }
+
+        private LambdaExpression? _expression;
+        /// <summary>
+        /// Gets or sets a delegate in the form of a <see cref="LambdaExpression"/> to be used to populate
+        /// this method argument while building the field resolver.
+        /// <br/><br/>
+        /// If not set, a query argument will be added to the field and the argument's value will be used
+        /// to populate the method argument while building the field resolver.
+        /// <br/><br/>
+        /// The delegate must of the type
+        /// <see cref="Expression{TDelegate}">Expression</see>&lt;<see cref="Func{T, TResult}">Func</see>&lt;<see cref="IResolveFieldContext"/>, TReturnType&gt;&gt;
+        /// where TReturnType matches <see cref="ParameterInfo">ParameterInfo</see>.<see cref="ParameterInfo.ParameterType">ParameterType</see>.
+        /// </summary>
+        public LambdaExpression? Expression
+        {
+            get => _expression;
+            set
+            {
+                if (value != null && (value.ReturnType != ParameterInfo.ParameterType || value.Parameters.Count != 1 || value.Parameters[0].Type != typeof(IResolveFieldContext)))
+                {
+                    throw new ArgumentException($"Value must be a lambda expression delegate of type Func<IResolveFieldContext, {ParameterInfo.ParameterType.Name}>.");
+                }
+                _expression = value;
+            }
+        }
+
+        /// <summary>
+        /// Builds a query argument or expression from this instance.
+        /// <br/><br/>
+        /// If a query argument is returned, it will be added to field type.
```

Comment:
```suggestion
        /// If a query argument is returned, it will be added to arguments of field type.
```

Also it worth to note _where_ it will be added. I understand what you wrote but class/method are public so comment seems a bit strange.

Reply:
```suggestion
        /// If a query argument is returned, it will be added to the arguments list of the field type.
```

Reply:
I know it's a little odd, but it was written from the perspective of someone attempting to override the method, to help them understand what will happen when it is overridden.  I tried to be as clear as possible.

Reply:
I'm fine to keep as is or you may even extend your comment. As you see fit.